mopidy/core/playback.py

```diff
@@ -541,3 +541,25 @@ def _trigger_seeked(self, time_position):
         # TODO: Trigger this from audio events?
         logger.debug('Triggering seeked event')
         listener.CoreListener.send('seeked', time_position=time_position)
+
+    def _export_state(self):
+        """Internal method for :class:`mopidy.Core`."""
+        return models.PlaybackState(
+            tl_track=self.get_current_tl_track(),
+            position=self.get_time_position(),
+            state=self.get_state())
+
+    def _restore_state(self, state, coverage):
+        """Internal method for :class:`mopidy.Core`."""
+        if state:
+            if not isinstance(state, models.PlaybackState):
+                raise TypeError('Expect an argument of type "PlaybackState"')
+            new_state = ''
+            if 'play-always' in coverage:
+                new_state = PlaybackState.PLAYING
+            if 'play-last' in coverage:
+                new_state = state.state
+            if state.tl_track is not None:
+                if PlaybackState.PLAYING == new_state:
+                    self.play(tl_track=state.tl_track)
+                    # TODO: seek to state.position?
```

Comment:
There are some challenges to seeking correctly in this case, as all of the audio system is async. But I think I have a lot of these solved due to the gapless fixing. So this might be trivial in the future.